elisp/intero.el

```diff
@@ -802,19 +801,37 @@ CHECKER and BUFFER are added to each item parsed from STRING."
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; Traditional completion-at-point function
 
-(defun intero-completion-at-point ()
+(defun intero-repl-completion-at-point ()
   "A (blocking) function suitable for use in `completion-at-point-functions'."
-  (let ((prefix-info (intero-completions-grab-prefix)))
+  (let ((prefix-info (intero-text-from-prompt-to-point)))
     (when prefix-info
       (cl-destructuring-bind
           (beg end prefix _type) prefix-info
-        (let ((completions
-               (intero-completion-response-to-list
-                (intero-blocking-call
-                 'backend
-                 (format ":complete repl %S" prefix)))))
-          (when completions
-            (list beg end completions)))))))
+        (let* ((tmp-buf (get-buffer-create (make-temp-name "intero-completions")))
```

Comment:
You probably want `with-temp-buffer`, rather than to create a buffer with a temporary name you've generated yourself. (`make-temp-name` is geared towards temporary filenames, anyway, whereas temp buffers typically look like ` *some-temp-thing*`: note the leading space, which excludes it from the buffer list.)

Reply:
Ah. `comint-redirect-send-command` uses the current buffer's process so we need to hold on to a reference to the intero buffer process and call `comint-redirect-send-command-to-process`. My quick fix didn't work so I need think more about nested let bindings.

Reply:
Yep, sorry, didn't mention that. :-)